src/renderers/shared/stack/reconciler/ReactCompositeComponent.js

```diff
@@ -104,6 +70,16 @@ function isPureComponent(Component) {
   return !!(Component.prototype && Component.prototype.isPureReactComponent);
 }
 
+// Separated into a function to contain deoptimizations caused by try/finally.
+function measureLifeCyclePerf(fn, debugID, timerType) {
+  ReactInstrumentation.debugTool.onBeginLifeCycleTimer(debugID, timerType);
+  try {
+    return fn();
+  } finally {
```

Comment:
Today I learned that you can use try without catching!

<img width="513" alt="screen shot 2016-08-23 at 2 31 17 pm" src="https://cloud.githubusercontent.com/assets/197597/17910596/60c6eb3a-693e-11e6-8762-2e2185c7d11b.png">